meta-balena-common/recipes-containers/hostapp-update/files/hostapp-update

```diff
@@ -56,6 +56,10 @@ mkdir -p /mnt/sysroot
 for dir in 'dev' 'etc' 'balena' 'hostapps' 'mnt/state' 'proc' 'sbin' 'sys' 'tmp'; do
 	mkdir -p "$SYSROOT/$dir"
 done
+if [ -d "/mnt/data/docker/tmp" ] && [ ! -L "$SYSROOT/balena/tmp" ]; then
+	rm -fr "$SYSROOT/balena/tmp"
+	ln -s /mnt/data/docker/tmp/ "$SYSROOT/balena/tmp"
+fi
```

Comment:
This patch does work around the problem. But it does depend on the fact that there is space in the data partition.

I would like to suggest a more flexible approach instead.

Add another argument to the script. Something like `--tmp-folder-symlink`

If that is passed, `"$SYSROOT/balena/tmp"` is symlinked to that location as long as it is a directory.

So we'll have a sensible default which is `"$SYSROOT/balena/tmp"`

And if possible, a user of hostapp-update can override it.

Keep in mind, at the moment, `"$SYSROOT/balena/tmp"` is made using `mkdir -p` in the loop above. There will need to be a cleanup if there is a symlink in that location.

Reply:
@ZubairLK f83b688 should implement your suggestions above (this has not been tested yet)

Reply:
I am wondering whether specifying a DOCKER_TMPDIR to the balena engine host instance is a better way than symlinking this.